lib/registry.js

```diff
@@ -29,48 +34,99 @@ export default class Registry {
     this.providers.add(provider)
     return provider
   }
-  getTitles(): Array<string> {
-    let texts = []
-    for (const provider of this.providers) {
-      texts = texts.concat(Array.from(provider.texts))
+  statusAdd(provider: Provider, status: { title: string, priority: number }): void {
+    for (let i = 0; i < this.itemsActive.length; i++) {
+      const entry = this.itemsActive[i]
+      if (entry.title === status.title && entry.provider === provider) {
+        // Item already exists, ignore
+        break
+      }
     }
-    return texts.sort(function(a, b) {
-      return a.priority - b.priority
-    }).map(function(item) {
-      return item.title
+
+    this.itemsActive.push({
+      title: status.title,
+      priority: status.priority,
+      provider,
+      timeAdded: Date.now(),
+      timeRemoved: null,
     })
+    this.emitter.emit('did-update')
+  }
+  statusRemove(provider: Provider, title: string): void {
+    for (let i = 0; i < this.itemsActive.length; i++) {
+      const entry = this.itemsActive[i]
+      if (entry.provider === provider && entry.title === title) {
+        this.pushIntoHistory(i, entry)
+        this.emitter.emit('did-update')
+        break
+      }
+    }
   }
-  getHistory(): Array<string> {
-    const uniqueHistory = []
-    const currentTitles = this.getTitles()
-    const history = this.history.concat(currentTitles)
-    for (let i = 0, length = history.length; i < length; ++i) {
-      if (history.lastIndexOf(history[i]) === i) {
-        uniqueHistory.push(history[i])
+  statusClear(provider: Provider): void {
+    let triggerUpdate = false
+    for (let i = 0; i < this.itemsActive.length; i++) {
+      const entry = this.itemsActive[i]
+      if (entry.provider === provider) {
+        this.pushIntoHistory(i, entry)
+        triggerUpdate = true
+        i--
       }
     }
-    this.history = uniqueHistory.slice(-50)
-    if (this.itemsToShowInHistory === 0) {
-      return []
+    if (triggerUpdate) {
+      this.emitter.emit('did-update')
+    }
+  }
+  pushIntoHistory(index: number, item: Signal): void {
+    item.timeRemoved = Date.now()
+    this.itemsActive.splice(index, 1)
+    this.itemsHistory = this.itemsHistory.concat([item]).slice(-1000)
+  }
+  getActiveTitles(): Array<string> {
+    return this.itemsActive.slice().sort(function(a, b) {
+      return a.priority - b.priority
+    }).map(i => i.title)
+  }
+  getOldTitles(): Array<{ title: string, duration: string }> {
+    const toReturn = []
+    const history = this.itemsHistory
+    const activeTitles = this.getActiveTitles()
+    const mergedTogether = history.map(i => i.title).concat(activeTitles)
+
+    for (let i = 0, length = history.length; i < length; i++) {
+      const item = history[i]
+      if (mergedTogether.lastIndexOf(item.title) === i) {
+        toReturn.push({
+          title: item.title,
+          duration: humanizeTime(item.timeRemoved && item.timeRemoved - item.timeAdded),
```

Comment:
Running this through `humanize-time` makes it virtually useless... I have yet to see an item show anything other than `0 s`.